File: solutions/java/MinimumDistances.java
import java.util.HashMap;

public class MinimumDistances{
	public static int run(int[] a){
		HashMap<Integer, Integer> valuesChecked = new HashMap<Integer, Integer>();
		int minDist = -1;
		int dist = minDist;

		for(int i = 0; i < a.length; i++){
			if(valuesChecked.containsKey(a[i])){
				dist = i - valuesChecked.get(a[i]);
				if(minDist == -1 || minDist > dist){
					minDist = dist;
				}
			}
			valuesChecked.put(a[i], i);
		}
		return minDist;
	}
}

File: solutions/kotlin/MinimumDistances.kt
fun minimumDistances(a: Array<Int>): Int {
    val valuesChecked = HashMap<Int, Int>()
    var minDist = -1
    var dist = minDist
    
    for(i in 0..(a.size - 1)){
        if(valuesChecked.containsKey(a[i])){
            dist = i - (valuesChecked.get(a[i]) ?: 0)
            if(minDist == -1 || minDist > dist){
                minDist = dist;
            }
        }
        valuesChecked.put(a[i], i)
    }
    return minDist
}